### PR TITLE
gsagh-247 move reaction force vector class into goo class

### PR DESCRIPTION
**Pull request by DominikaLos**

### CODE REVIEW

File: GsaGH/Parameters/5_Results/PointResultGoo.cs [hunk 1]
```diff
@@ -1,48 +1,40 @@
...
-    public PointResultGoo(Point3d point, IQuantity result, Color colour, float size)
+    public PointResultGoo(Point3d point, IQuantity result, Color color, float size)
     : base(point)
     {
-      m_result = result;
-      m_size = size;
-      m_colour = colour;
+      MResult = result;
+      _mSize = size;
+      _mColor = color;
```
Comment (kpne) [inline GsaGH/Parameters/5_Results/PointResultGoo.cs:17]:
```suggestion
      Result = result;
      _size = size;
      _colour = color;
```

File: GsaGH/Parameters/5_Results/PointResultGoo.cs [hunk 2]
```diff
@@ -1,48 +1,40 @@
...
-    }
-    public override string TypeName
-    {
-      get { return "Result Point"; }
-    }
-    public override string TypeDescription
-    {
-      get { return "A GSA result point type."; }
-    }
+    internal IQuantity MResult;
```
Comment (kpne) [inline GsaGH/Parameters/5_Results/PointResultGoo.cs:23]:
```suggestion
    internal IQuantity Result;
```

File: GsaGH/Parameters/5_Results/PointResultGoo.cs [hunk 3]
```diff
@@ -1,48 +1,40 @@
...
+      MResult = result;
+      _mSize = size;
+      _mColor = color;
     }
 
-    internal IQuantity m_result;
-    private float m_size;
-    private Color m_colour;
+    private readonly float _mSize;
+    private readonly Color _mColor;
```
Comment (kpne) [inline GsaGH/Parameters/5_Results/PointResultGoo.cs:21]:
```suggestion
    private readonly float _size;
    private readonly Color _colour;
```

**Review by kpne — CHANGES_REQUESTED** [review]

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 4]
```diff
@@ -13,68 +13,51 @@ namespace GsaGH.Parameters
   /// </summary>
   public class LineResultGoo : GH_GeometricGoo<Line>, IGH_PreviewData
   {
-    public LineResultGoo(Line line, IQuantity result1, IQuantity result2, Color colour1, Color colour2, float size1, float size2) : base(line)
+    public LineResultGoo(Line line, IQuantity result1, IQuantity result2, Color color1, Color color2, float size1, float size2) : base(line)
```
[the remark this answers is not in the document]
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:16]: didn't we decide to use "colour" only in UI part? ;)

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 5]
```diff
@@ -13,68 +13,51 @@ namespace GsaGH.Parameters
...
-      previewResultThk.Add((int)Math.Abs(((m_size2 - m_size1) * 1 + m_size1)));
-      previewResultSegments.Add(new Line(Value.PointAt(0.5), Value.PointAt(1)));
+      Result1 = result1;
+      Result2 = result2;
+      _size1 = size1;
+      _size2 = size2;
+      _color1 = color1;
+      _color2 = color2;
+
+      var gHGradient = new Grasshopper.GUI.Gradient.GH_Gradient();
```
[the remark this answers is not in the document]
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:25]: didn't we decide to go with var when type is obvious?

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 6]
```diff
@@ -13,68 +13,51 @@ namespace GsaGH.Parameters
...
-
-      previewResultColours.Add(gH_Gradient.ColourAt(1));
-      previewResultThk.Add((int)Math.Abs(((m_size2 - m_size1) * 1 + m_size1)));
-      previewResultSegments.Add(new Line(Value.PointAt(0.5), Value.PointAt(1)));
+      Result1 = result1;
+      Result2 = result2;
+      _size1 = size1;
+      _size2 = size2;
+      _color1 = color1;
+      _color2 = color2;
```
[the remark this answers is not in the document]
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:23]: as above

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 7]
```diff
@@ -13,68 +13,51 @@ namespace GsaGH.Parameters
...
-    internal List<int> previewResultThk;
-
-    public override string ToString()
-    {
-      return string.Format("LineResult: L:{0:0.0}, R1:{1:0.0}, R2:{2:0.0}", Value.Length, m_result1, m_result2);
-    }
+    private readonly float _size1;
+    private readonly float _size2;
+    private readonly Color _color1;
+    private readonly Color _color2;
```
[the remark this answers is not in the document]
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:44]: as above :)

File: GsaGH/Parameters/5_Results/LineResultGoo.cs [hunk 8]
```diff
@@ -141,14 +121,11 @@ public override bool CastFrom(object source)
         return true;
       }
 
-      Line line = new Line();
-      if (GH_Convert.ToLine(source, ref line, GH_Conversion.Both))
-      {
-        Value = line;
-        return true;
-      }
+      var line = new Line();
```
[the remark this answers is not in the document]
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:124]: as above

File: GsaGH/Parameters/5_Results/PointResultGoo.cs (hunk 1 above)
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/PointResultGoo.cs:17]: result and size ok, but color or colour?? XD

File: GsaGH/Parameters/5_Results/PointResultGoo.cs (hunk 3 above)
Comment (DominikaLos) [reply GsaGH/Parameters/5_Results/PointResultGoo.cs:21]: as above :)

Comment (kpne) [reply GsaGH/Parameters/5_Results/PointResultGoo.cs:21]: review was from before our chat, agree to use `_color` in code

File: GsaGH/Parameters/5_Results/PointResultGoo.cs (hunk 1 above)
Comment (kpne) [reply GsaGH/Parameters/5_Results/PointResultGoo.cs:17]: review was from before our chat, agree to use _color in code

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:124]: review was from before our chat, agree to use _color in code

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:23]: review was from before our chat, agree to use _color in code

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:25]: review was from before our chat, agree to use google style guide encouraged use of `var`

File: GsaGH/Parameters/5_Results/LineResultGoo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Parameters/5_Results/LineResultGoo.cs:16]: review was from before our chat, agree to `color` in code

File: GsaGH/Parameters/5_Results/MeshResultGoo.cs [hunk 9]
```diff
@@ -28,37 +29,33 @@ public Mesh ValidMesh
...
 
-        for (int i = 0; i < ngons.Count; i++)
+        foreach (var faceId in ngons.Select(ngon => ngon.FaceIndexList().Select(u => (int)u).ToList()).SelectMany(faceIndex => faceIndex))
         {
-          List<int> faceindex = ngons[i].FaceIndexList().Select(u => (int)u).ToList();
-          for (int j = 0; j < faceindex.Count; j++)
-          {
-            m.Faces.AddFace(x.Faces[faceindex[j]]);
-          }
+          m.Faces.AddFace(x.Faces[faceId]);
```
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Parameters/5_Results/MeshResultGoo.cs:42]: resolving this as it is a suggestion not a required change

**Review by kpne — APPROVED** [review]

**Comment by kpne** [comment]
@DominikaLos the pipeline now fails to build the project, I think you might have to apply the suggestions consistently locally

**Pull request merged**